### PR TITLE
[timeline] Fix more issues

### PR DESCRIPTION
**Pull request by chejennifer**
- add click target for legend [issue #1034](https://github.com/datacommonsorg/website/issues/1034)
- disable chart tooltips past the chart area [issue #1035](https://github.com/datacommonsorg/website/issues/1035)
- fix an issue with tooltips in the timeline tool for when a chart has multiple stat vars and multiple places but some places don't have data for one of those stat vars. In these cases, the tooltip doesn't specify which stat var the data is for. To see this bug, go [here](https://autopush.datacommons.org/tools/timeline#place=geoId%2F0606000%2CgeoId%2F0656938%2Ccountry%2FUSA&statsVar=Count_Person__BelowPovertyLine__Count_Person_PerArea__GrowthRate_Count_Person) and try hovering over around 2016.

dev updated

### CODE REVIEW

**Comment by shifucun** [comment]
Left some comments in https://github.com/datacommonsorg/website/issues/1034#issuecomment-881768696. I do feel it needs more thinking on where these links go

**Review by beets — COMMENTED** [review]
lgtm - re: the comment from bo, replied to the issue

File: static/js/chart/draw.ts [hunk 1]
```diff
@@ -166,36 +166,47 @@ function getTooltipContent(
   let tooltipDate = "";
   let tooltipContent = "";
   const places = Object.keys(dataGroupsDict);
+  const dataGroupLabels: Set<string> = new Set();
   for (const place of places) {
-    const dataGroups = dataGroupsDict[place];
-    for (const dataGroup of dataGroups) {
+    dataGroupsDict[place].forEach((dataGroup) =>
```
[the remark this answers is not in the document]
Comment (chejennifer) [reply static/js/chart/draw.ts]: done

**Review by beets — APPROVED** [review]
thanks for the updates!

**Pull request merged**